app/controllers/calagator/venues_controller.rb

```diff
@@ -7,24 +7,34 @@ class VenuesController < Calagator::ApplicationController
   include DuplicateChecking::ControllerActions
   require_admin only: [:duplicates, :squash_many_duplicates]
 
+  def venue
+    @venue ||= params[:id] ? Venue.find(params[:id]) : Venue.new
+  end
+
+
   # GET /venues
-  # GET /venues.xml
   def index
     @search = Venue::Search.new(params.permit!)
     @venues = @search.venues
 
     flash[:failure] = @search.failure_message
     return redirect_to venues_path if @search.hard_failure?
+    render_venues @venues
+  end
 
+  def render_venues venues
```

Comment:
Yeah, I've been on the fence about whether this particular action is complicated enough to justify a class extraction. I did try it, but ended up going back to the simpler private method extraction. The class felt just a little too heavy. Very subjective, though, of course.